### PR TITLE
Remove parsing error when json request has empty body

### PR DESCRIPTION
**Pull request by DiogoMRSilva**
Closes https://github.com/zaproxy/zaproxy/issues/7174

### CODE REVIEW

File: zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java [hunk 1]
```diff
@@ -74,6 +74,9 @@ public boolean isValidContentType(String contentType) {
     @Override
     public void parseContent(String content) {
         sr = new SimpleStringReader(content);
+        if (content == null || content.length() == 0 || content.strip().length() == 0) {
```
Comment (kingthorin) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java:77]: isEmpty()

**Comment by kingthorin** [comment]
Looks like `strip` is a Java 11 feature?

File: zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java (hunk 1 above)
Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java:77]: And is the strip really needed?

Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java:77]: `content` can't be null here.

Comment (DiogoMRSilva) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java:77]: will fix it

**Comment by DiogoMRSilva** [comment]
> Looks like `strip` is a Java 11 feature?

yes I think so, will remove it

File: zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java (hunk 1 above)
Comment (DiogoMRSilva) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java]: replaced the whole check with `isBlank()` and did put the `sr` instantiating after the check.

Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java]: That method is also Java 11.

Comment (kingthorin) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java]: To stay Java 8 compatible you'll need to use apache lang or lang3 `StringUtils.isBlank(content)`

Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java]: Better the latter.

Comment (DiogoMRSilva) [reply zap/src/main/java/org/parosproxy/paros/core/scanner/VariantJSONQuery.java]: So going to leave it with only the `isEmpty` check.

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**